### PR TITLE
Add a default of 3 retries for uploading a file

### PR DESCRIPTION
**Pull request by mendsley**

### CODE REVIEW

File: src/retry.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	constructor() {
+		this.maxAttemps = maxRetries
+	}
+
+	async execute<T>(action: () => Promise<T>): Promise<T> {
+		// run N-1 attempts
+		var attempt = 1;
+		while (attempt < this.maxAttemps) {
+			try {
+				return await action()
```
Comment (jsmrcina) [inline src/retry.ts:17]: Was just gonna ask about the empty try/catch :slightly_smiling_face:

File: src/retry.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+			} catch (err) {
+				core.info( (err as any)?.message )
+			}
+
+			// sleep before retrying
+			const seconds = Math.random() * 10 + 5
```
Comment (jsmrcina) [inline src/retry.ts:23]: nit: Maybe do exponential backoff here since it's calling into a web service

**Review by jsmrcina — APPROVED** [review]

File: src/retry.ts (hunk 1 above)
Comment (mendsley) [reply src/retry.ts:17]: Got too ahead of myself!

File: src/retry.ts (hunk 2 above)
Comment (mendsley) [reply src/retry.ts:23]: That's a great idea. I'll add that as a second PR to this

**Pull request merged**